File: 03Builder/Students/mkacprzak/Builder/src/main/java/FluentBuilder/SimpleFluentBuilder/Computer.java
package FluentBuilder.SimpleFluentBuilder;

import FluentBuilder.*;

public class Computer {

    private Type type;
    private GPU gpu;
    private CPU cpu;
    private Ram ram;
    private Motherboard motherboard;


public Computer type(Type type)
{
    this.type=type;
    return this;
}
    public Computer cpu(CPU cpu)
    {
        this.cpu=cpu;
        return this;
    }
    public Computer ram(Ram ram)
    {
        this.ram=ram;
        return this;
    }
    public Computer gpu(GPU gpu)
    {
        this.gpu=gpu;
        return this;
    }
    public Computer motherboard(Motherboard motherboard)
    {
        this.motherboard=motherboard;
        return this;
    }

    @Override
    public String toString() {
        return "Computer{" +
                "type=" + type +
                ", gpu=" + gpu +
                ", cpu=" + cpu +
                ", ram=" + ram +
                ", motherboard=" + motherboard +
                '}';
    }
}

File: 03Builder/Students/2019/KaczmarzTomasz/BuilderPattern/Computer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BuilderPattern
{
    enum ComputerType
    {
        GamingPC,
        OfficePC,
        MidRangePC
    }

    enum Part
    {
        CPU,
        GPU,
        Motherboard,
        RAM
    }

    class Computer
    {
        private ComputerType computerType;
        private Dictionary<Part, string> parts = new Dictionary<Part, string>();

        public Computer(ComputerType type)
        {
            computerType = type;
        }

        public string this[Part type]
        {
            get { return parts[type]; }
            set { parts[type] = value; }
        }

        public void Show()
        {
            Console.WriteLine("COMPUTER SPECS:");
            Console.WriteLine("\tType: " + computerType);
            Console.WriteLine("\tCPU: " + parts[Part.CPU]);
            Console.WriteLine("\tGPU: " + parts[Part.GPU]);
            Console.WriteLine("\tMotherboard: " + parts[Part.Motherboard]);
            Console.WriteLine("\tRAM: " + parts[Part.RAM]);
        }
    }
}
